src/vma/dev/net_device_val.cpp

```diff
@@ -212,9 +212,10 @@ net_device_val::net_device_val(void *desc) : m_lock("net_device_val lock")
 	m_rdma_key = cma_id->route.addr.addr.ibaddr.pkey;
 
 	set_ifname(ifa->ifa_name);
-	set_if_idx(if_nametoindex(m_name.c_str()));
+	set_if_idx(if_nametoindex(ifa->ifa_name));
+	set_type(get_iftype_from_ifname(ifa->ifa_name));
 	set_flags(ifa->ifa_flags);
-	set_mtu(get_if_mtu_from_ifname(m_name.c_str()));
```

Comment:
There are two more locations in the constructor which uses m_name.c_str(). do we want to replace them ?
Why the type of m_name is string ? please consider change it to char*